### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by lfrancke**
Create a span for each new reconcile, assign a unique id to each request and improve logging in general

### CODE REVIEW

File: src/controller.rs [hunk 1]
```diff
@@ -243,8 +244,12 @@ where
             .run(reconcile, error_policy, context)
             .for_each(|res| async move {
                 match res {
-                    Ok(o) => info!("Reconciled {:?}", o),
-                    Err(ref e) => error!("Reconcile failed: {:?}", e),
+                    Ok(o) => trace!(resource = ?o, "Reconciliation finished successfully (it is normal to see this message twice)"),
+                    Err(ref err) => {
+                        // If we get here it means that our `reconcile` method returned an error which should never happen because
+                        // we convert all errors to requeue operations.
+                        error!(?err, "Reconciliation finished with an error, this should not happen, please file an issue")
```
Comment (soenkeliebau) [inline src/controller.rs:251]: Random brainfart, we are calling for people to file issues in some places, should we keep track of these places somewhere and give them an identifier that people can reference or something similar?
Nothing to do about this here, just a thought to be discussed.

**Review by soenkeliebau — APPROVED** [review]
LGTM - just one note that we might think about but that is just because it occurred to me here

**Comment by lfrancke** [comment]
Oh, I'm sorry, not sure why I added you as a reviewer. This is still a draft but it looks good so it can probably go in as well.

File: src/controller.rs (hunk 1 above)
Comment (lfrancke) [reply src/controller.rs:251]: Yeah...we should reactivate our common logging thing but that's probably very much outdated by now. It's a good idea though.

**Pull request merged**